compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -529,6 +529,32 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             obligation.predicate.to_poly_trait_ref(),
             trait_ref,
         )?);
+
+        let lang_items = self.tcx().lang_items();
+        if [lang_items.fn_once_trait(), lang_items.fn_trait(), lang_items.fn_mut_trait()]
+            .contains(&Some(obligation.predicate.def_id()))
+            // Skip `MiscObligation`s for better output.
+            && matches!(obligation.cause.code, ObligationCauseCode::BindingObligation(_, _))
```

Comment:
You can look at the diff between the two commits to see what this check changes.